### PR TITLE
Ring menu item

### PR DESCRIPTION
**Pull request by EstevanBR**

### CODE REVIEW

File: RingMenu/RingMenu.cpp [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
+    rad = 0.0f;
     this->setInfo(std::string("This is a Ring Menu"));
-    this->setPosition(ScreenWidth/2,ScreenHeight/2);
-    this->setRadius(ScreenHeight/8);
-    this->move(-ScreenHeight/8,-ScreenHeight/8);
+
+    //this->setOrigin(radius/2, radius/2);
+    this->setPosition(ScreenManager::screenWidth()/2,ScreenManager::screenHeight()/2);
+    this->setRadius(radius);
+    this->move(-ScreenManager::screenHeight()/8,-ScreenManager::screenHeight()/8);
```
Comment (gillett-hernandez) [inline RingMenu/RingMenu.cpp]: you seem to have a double screenHeight call here

File: RingMenu/RingMenu.cpp [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
     this->setPointCount(3);
     this->setOutlineColor(sf::Color(ringColor));
     this->setFillColor(clearColor);
     this->setOutlineThickness(1);
-    this->hidden = true;
+    //this->hidden = true;
+    RingMenuItem item = RingMenuItem(this, RingMenuItemTypeWeapon);
+    RingMenuItem armor = RingMenuItem(this, RingMenuItemTypeArmor);
+    RingMenuItem shield = RingMenuItem(this, RingMenuItemTypeItem);
+    this->items.push_back(item);
```
Comment (gillett-hernandez) [inline RingMenu/RingMenu.cpp]: noice using vectors and stuff

File: RingMenu/RingMenu.cpp [hunk 3]
```diff
@@ -86,25 +97,40 @@ void RingMenu::up() {
...
 }
 void RingMenu::left() {
     if (debug) {
         std::cout << "RingMenu::left" << std::endl;
     }
-    this->move(-5, 0);
+
+    rad -= 0.05f;
+
+    for (size_t i = 0; i < this->items.size(); i++) {
```
Comment (gillett-hernandez) [inline RingMenu/RingMenu.cpp]: im sure its simple but why do you change the rad and then set all the item positions each time RingMenu::left is called

File: RingMenu/RingMenu.cpp [hunk 4]
```diff
@@ -86,25 +97,40 @@ void RingMenu::up() {
...
 }
 void RingMenu::right() {
     if (debug) {
         std::cout << "RingMenu::right" << std::endl;
     }
-    this->move(5, 0);
+
+    rad += 0.05f;
+
+    for (size_t i = 0; i < this->items.size(); i++) {
```
Comment (gillett-hernandez) [inline RingMenu/RingMenu.cpp]: ditto here

File: RingMenu/RingMenu.cpp [hunk 5]
```diff
@@ -117,6 +143,9 @@ void RingMenu::enterPressed() {
 
 void RingMenu::drawToWindow(sf::RenderWindow &windowRef) {
     windowRef.draw(*this);
+    for (RingMenuItem item: this->items) {
```
Comment (gillett-hernandez) [inline RingMenu/RingMenu.cpp]: noice, subitems and stuff

File: RingMenu/RingMenuItem.cpp [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    this->setTexture(TextureManager::instance()->getRef(kRingMenuItemTexture));
+    this->setType(RingMenuItemTypeItem);
+}
+RingMenuItem::RingMenuItem(RingMenu* parent, RingMenuItemType type) {
+    this->parent = parent;
+    //this->setColor(sf::Color(0xFF, 0xFF, 0xFF, 0xFF));
+    //this->setPosition(100, 100);
+    //float v = this->parent->getRadius() - this->parent->getPosition().x;
+    // for some reason 23 seems to be the magic value. 2 minutes later - its because 24.04 = 17*sqrt(2)
+    float origin = (this->parent->getPosition().x + this->parent->getRadius())/sqrt(2) - this->parent->getPosition().x - 24.04;
```
Comment (gillett-hernandez) [inline RingMenu/RingMenuItem.cpp]: maybe calculate the magic number from the other numbers. 24.04 = 17 * sqrt(2) but where does 17 come from. ive forgotten

File: RingMenu/RingMenuItem.cpp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    std::cout << type;
+}
+void RingMenuItem::setType(RingMenuItemType type) {
+    this->type = type;
+    switch (this->type) {
+        case RingMenuItemTypeNone: {
+            break;
+        }
+        case RingMenuItemTypeArmor: {
+            this->setTextureRect(sf::IntRect(34*4, 34*5, 34, 34));
```
Comment (gillett-hernandez) [inline RingMenu/RingMenuItem.cpp]: these too. you should define 34 as the texture size(`#define texture_size 34`) or something. or maybe add a method that takes say (4,5) and sets the texture rect from that.

File: RingMenu/RingMenu.cpp (hunk 3 above)
Comment (EstevanBR) [inline RingMenu/RingMenu.cpp]: its so when you're holding down left, the menu items rotate left/right.

File: RingMenu/RingMenuItem.cpp (hunk 6 above)
Comment (EstevanBR) [inline RingMenu/RingMenuItem.cpp]: 17 is the radius

File: RingMenu/RingMenuItem.cpp (hunk 7 above)
Comment (EstevanBR) [inline RingMenu/RingMenuItem.cpp]: so the texture I'm using here is just temporary / placeholder. I can make some methods but the value of 34 will need to be changed depending on which image file is being used.

**Comment by EstevanBR** [comment]
@gillett-hernandez I've implemented the changes

File: RingMenu/RingMenu.cpp (hunk 1 above)
Comment (EstevanBR) [inline RingMenu/RingMenu.cpp]: this is fixed

File: RingMenu/RingMenuItem.cpp (hunk 6 above)
Comment (EstevanBR) [inline RingMenu/RingMenuItem.cpp]: you know what this value isn't right anyways.

(34 / 2pi) * sqrt(2) looks a lot better

**Pull request merged**